Tests/Runner/BranchedWorkspaceTest.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+namespace Keboola\DockerBundle\Tests\Runner;
+
+use Keboola\Csv\CsvFile;
+use Keboola\DockerBundle\Docker\Component;
+use Keboola\DockerBundle\Docker\JobDefinition;
+use Keboola\DockerBundle\Docker\Runner;
+use Keboola\DockerBundle\Docker\Runner\UsageFile\NullUsageFile;
+use Keboola\DockerBundle\Tests\BaseRunnerTest;
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\DevBranches;
+use Keboola\StorageApi\Options\Components\Configuration;
+use Keboola\StorageApiBranch\ClientWrapper;
+use Keboola\Temp\Temp;
+
+class BranchedWorkspaceTest extends BaseRunnerTest
+{
+    const BRANCH_NAME = 'workspace-test';
+    const BUCKET_NAME = 'workspace-test';
+    const TABLE_NAME = 'test-table';
+
+    /**
+     * @doesNotPerformAssertion
+     */
+    public function testConfigIsAvailableToJobWhenCreatedBeforeBranch()
+    {
+        $storageApi = $this->getClient();
+        $storageApiMaster = $this->getMasterClient();
+
+        // prepare data
+        $inBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_IN);
+        $outBucketId = $this->cleanupAndCreateBucket($storageApi, self::BUCKET_NAME, $storageApi::STAGE_OUT);
+
+        $this->loadDataIntoTable($storageApi, $inBucketId, 'test-table', [
+            ['id', 'text'],
+            ['test1', 'test1'],
+        ]);
+
+        // setup configuration outside branch
+        $componentsApiForConfig = new Components($storageApi);
+        $configId = $this->generateConfigId();
+
+        $this->cleanupAndCreateConfiguration(
+            $componentsApiForConfig,
+            (new Configuration())
+                ->setComponentId('keboola.runner-workspace-test')
+                ->setName($configId)
+                ->setConfigurationId($configId)
+        );
+
+        // create branch
+        $branchesApiMaster = new DevBranches($storageApiMaster);
+        $branchId = $this->cleanupAndCreateBranch($branchesApiMaster, self::BRANCH_NAME);
+        $storageApiWrapper = new ClientWrapper($storageApi, null, null, $branchId);
+
+        // run testing job
+        $this->runCopyJob($storageApiWrapper, $configId, $inBucketId, $outBucketId, self::TABLE_NAME);
+    }
+
+    public function testConfigIsNotAvailableToJobWhenCreatedAfterBranch()
```

Comment:
přemýšlím k čemu je nám tenhle test a nic mě nenapadá. Vyhodil bych ho.

Reply:
Testoval jsme si tim to chovani, ktere si mi popisoval no a kdyz uz jsme to mel napsane, bylo im lito ho mazat :D 

Kazdopadne, to ze job v branch failne, pokud chce pouzit main config, ktery byl zalozeny az po zalozeni branch je zname/dokumentovane chovani nebo nejaky haluz side-effect kteryho se chceme zbavit? V tom prvnim pripade by mi totiz davalo smysl test zachovat, protoze na to muze naopak nekdo spolehat.

Reply:
Odstraneno dle domluvy